microgateway_decorator/bin/compile

```diff
@@ -98,8 +98,32 @@ fi
 
 export PATH=$PATH:${BUILD_DIR}/tmp/node/bin
 # npm --prefix $BUILD_DIR/tmp/edgemicro install edgemicro
-git clone https://github.com/apigee-internal/microgateway.git $BUILD_DIR/tmp/edgemicro
-(cd $BUILD_DIR/tmp/edgemicro && git checkout tags/v2.5.8)
+
+# local installation if microgateway folder exists in lib directory
+if [ -d "$ROOT_DIR/lib/microgateway" ]; then
+  echo "Installing Edge Microgateway from lib directory."
+  cp -a $ROOT_DIR/lib/microgateway/. $BUILD_DIR/tmp/edgemicro
+else #clone from github
+  echo "Cloning Edge Microgateway from Github."
+  git clone https://github.com/apigee-internal/microgateway.git $BUILD_DIR/tmp/edgemicro
+fi
+
+# the envrionment variable is not a zero length string
+if [ ! -z "$APIGEE_MICROGATEWAY_VERSION" ]; then
+  echo "Trying git checkout tags/$APIGEE_MICROGATEWAY_VERSION"
+  CHECKOUT_RESULT=`(cd $BUILD_DIR/tmp/edgemicro && git checkout tags/$APIGEE_MICROGATEWAY_VERSION)`
+  if [ $? -ne 0  ]; then # version not found then prepend v to tag
+    echo "Trying git checkout tags/v$APIGEE_MICROGATEWAY_VERSION instead."
+    CHECKOUT_RESULT=`(cd $BUILD_DIR/tmp/edgemicro && git checkout tags/v$APIGEE_MICROGATEWAY_VERSION)`
+    if [ $? -ne 0  ]; then # fail the compile step
+      echo "The Edgemicro tag $APIGEE_MICROGATEWAY_VERSION does not exist." >&2
+      exit 1
+    fi
+  fi
+else # using most current version of Edge Microgateway.
+  echo "Not executing git checkout; using most current version in git repository instead."
+fi
+
 npm --prefix $BUILD_DIR/tmp/edgemicro install
 
 mkdir -p $BUILD_DIR/microgateway
```

Comment:
Let's go ahead and rename this to something like "decorator" to avoid conflicts with users uploading microgateway locally.

Reply:
This will also need to be updated in the `lib/microgateway.sh` and `lib/microgateway.py`

Reply:
So the plan is to change this code: 
`
mkdir -p $BUILD_DIR/microgateway
cp $ROOT_DIR/lib/microgateway.js $BUILD_DIR/microgateway
cp $ROOT_DIR/lib/package.json $BUILD_DIR/microgateway
npm --prefix $BUILD_DIR/microgateway install
`
to 
`mkdir -p $BUILD_DIR/decorator
cp $ROOT_DIR/lib/microgateway.js $BUILD_DIR/decorator
cp $ROOT_DIR/lib/package.json $BUILD_DIR/decorator
npm --prefix $BUILD_DIR/decorator install
`

Reply:
Is this correct?

Reply:
Yep :+1: